collect_app/src/main/java/org/odk/collect/android/preferences/FormManagementPreferences.java

```diff
@@ -145,4 +168,22 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
         });
     }
 
+    @Override
+    public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
+        if (key.equals(KEY_FORM_UPDATE_MODE) || key.equals(KEY_PERIODIC_FORM_UPDATES_CHECK)) {
+            formUpdateManager.scheduleUpdates();
```

Comment:
`SchedulerFormUpdateAndSubmitManager#scheduleUpdates` cancels any existing deferred work and then sets it back up again. The `if` here is just making sure we only do that when the relevant settings change (as this will fire for any changes).

Reply:
Ohhh, hah. Of course, that makes a lot more sense. I was going too quickly and interpreted it as if we go into match exactly or update downloaded modes.